packages/remix-dev/vite/plugin.ts

```diff
@@ -488,14 +483,19 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
     let { serverBuildFile, unstable_serverBundles: serverBundles } =
       resolvedRemixUserConfig;
 
-    let basename = resolvedRemixUserConfig.basename ?? "/";
+    let basename = resolvedRemixUserConfig.basename || "/";
     if (viteCommand === "serve" && !viteUserConfig.server?.middlewareMode) {
       // Odd restriction for default vite dev server
       // since Vite requires SSR request URL to be under `base` option (= `publicPath`)
-      invariant(
-        basename.startsWith(publicPath),
-        "'publicPath' must be a prefix of 'basename' for default Vite dev server."
-      );
+      if (!basename.startsWith(publicPath)) {
+        console.warn(
```

Comment:
I changed this from an `invariant` to a warning since the invariant ends up including a message like `"This is a bug in Remix"` in it